pkg/performanceprofile/cmd/render/render.go

```diff
@@ -210,6 +220,47 @@ func render(inputDir, outputDir string) error {
 	return nil
 }
 
+// isLegacySNOWorkloadPinningMethod provides a check for situations where the user is creating an SNO cluster with the
+// legacy method for CPU Partitioning. In order to make sure the bootstrap and running cluster MCs are synced up we check the MCs
+// provided by the user, if any one of them contain the file addition to `/etc/kubernetes/openshift-workload-pinning` and have not set
+// the API CPUPartitioningAllNodes at install time, we assume a legacy intention and alter the flag to generate the new bootstrap manifests.
+//
+// Note:
+//   - This will only trigger when Control plane is SNO or the CPU Partitioning API is NOT set to AllNodes
+//   - We do not alter the API flag here, when NTO starts up in cluster, it will notice the flag and
+//     update the flag and ignore the create error since the files already exist.
+func isLegacySNOWorkloadPinningMethod(mcs []*mcfgv1.MachineConfig, infra *apicfgv1.Infrastructure, partitioningMode *apicfgv1.CPUPartitioningMode) bool {
+
+	// If we can't determine SNO topology, we return.
+	if infra == nil {
+		return false
+	}
+
+	if infra.Status.ControlPlaneTopology != apicfgv1.SingleReplicaTopologyMode || (partitioningMode != nil && *partitioningMode == apicfgv1.CPUPartitioningAllNodes) {
+		return false
+	}
+
+	const kubernetesConfDir = "/etc/kubernetes/openshift-workload-pinning"
```

Comment:
How stable is this filename?

Reply:
Unless there's a redesign in what how kubelet handles that, this file name should be rather static. It's currently hardcoded https://github.com/openshift/kubernetes/blob/ba1825544533d273d86b405195ee791e500b74c7/pkg/kubelet/managed/managed.go#L31

Reply:
Stable. Path/filename is baked into kubelet

Reply:
Sure thing, updated with comment.